### PR TITLE
[Utils] Added decoding for image pixel geometry

### LINKED ISSUE #32: [Utility] Decode image pixel geometry
**Is your feature request related to a problem? Please describe.**
When requesting an object detection, a field called `geometry` is returned. This contains the pixel coordinates of the detection. For example, it contains the outline of a building which was detected in the photo with x,y coordinates in the image. It is encoded as a vector tile, so needs to be decoded in order to be used.

Example: https://graph.mapillary.com/1933525276802129/detections?fields=id,geometry&access_token=MLY|XXX

**Describe the solution you'd like**
A function should use the decode() method of mapbox-vector-tile library to convert the geometry field to a JSON. This would be a utility that can be used in another function, something like `get_detections(image_key)`.

**Describe alternatives you've considered**
This could also be a public method, but makes more sense as a utility that can be reused when getting information about the detections in either an image key or a map feature. 

**Additional context**
Detections are the segmentation in the image, where a class of object was detected, such as sidewalk, road, building, traffic sign, vegetation, or bench. Multiple detections are used to form point data, such as a bench with a longitude and latitude. In the API you can pass graph.mapillary.com/IMAGE_ID/detections to get a list of everything detected in an image, or graph.mapillary.com/MAP_FEATURE_ID/detections to get a list of all detections that were determined to be the same object from multiple angles, then combined to predict a point location. Users often want to get the detection pixel coordinates in order to draw this on top of a JPG or in the MapillaryJS image viewer library, to highlight a portion of the image that matches an object displayed on a map.

### PR DESCRIPTION
**Pull request by Rubix982**
# Pixel Geometry

:trophy: Added main utility `decode_pixel_geometry_in_geojson` for decoding all the pixel geometries from the GeoJSON.

This should close #32.

# Test Run
```python
import mapillary as mly
from utils.format import decode_pixel_geometry_in_geojson
mly.set_access_token('MLY|XXX')
data = mly.get_detections_with_image_id(image_id=1933525276802129).to_dict()
data = decode_pixel_geometry_in_geojson(geojson=data, normalized=True).to_dict() # for normalized coordinates
data = decode_pixel_geometry_in_geojson(geojson=data).to_dict() # for un-normalized coordinates
```

# Normalized output
```python
'type': 'FeatureCollection', 'features': [{'type': 'Feature', 'geometry': {'type': 'Point', 'coordinates': [-97.743279722222,
30.270651388889]}, 'properties': {'created_at': '2021-05-20T17:49:01+0000', 'id': '1942105415944115', 'image_id': None,
'pixel_geometry': {'coordinates': [[[0.610107421875, 0.37060546875], [0.615234375, 0.37060546875], [0.615234375,
0.361572265625], [0.610107421875, 0.361572265625], [0.610107421875, 0.37060546875]]]}, 'value': 'regulatory--no-
parking--g2'}}, {'type': 'Feature', 'geometry': {'type': 'Point', 'coordinates': [-97.743279722222, 30.270651388889]}, 'properties':
{'created_at': '2021-05-20T18:40:21+0000', 'id': '1942144785940178', 'image_id': None, 'pixel_geometry': {'coordinates':
[[[0.88818359375, 0.502685546875], [0.898681640625, 0.502685546875], [0.898681640625, 0.47509765625],
[0.88818359375, 0.47509765625], [0.88818359375, 0.502685546875]]]},
```

# Un-normalized output
```python
{'type': 'FeatureCollection', 'features': [{'type': 'Feature', 'geometry': {'type': 'Point', 'coordinates': [-97.743279722222,
30.270651388889]}, 'properties': {'created_at': '2021-05-20T17:49:01+0000', 'id': '1942105415944115', 'image_id': None,
'pixel_geometry': {'coordinates': [[[2499, 1518], [2520, 1518], [2520, 1481], [2499, 1481], [2499, 1518]]]}, 'value': 'regulatory 
--no-parking--g2'}}, {'type': 'Feature', 'geometry': {'type': 'Point', 'coordinates': [-97.743279722222, 30.270651388889]},
'properties': {'created_at': '2021-05-20T18:40:21+0000', 'id': '1942144785940178', 'image_id': None, 'pixel_geometry':
{'coordinates': [[[3638, 2059], [3681, 2059], [3681, 1946], [3638, 1946], [3638, 2059]]]}, 'value': 'information--parking--g1'}},
{'type': 'Feature', 'geometry': {'type': 'Point', 'coordinates': [-97.743279722222, 30.270651388889]},
```

### CODE REVIEW

File: mapillary/utils/format.py [hunk 1]
```diff
@@ -376,3 +379,205 @@ def geojson_to_polgyon(geojson: dict) -> dict:
             ],
         }
     )
+
+
+def flatten_dictionary(data: dict, parent_key: str = "", sep: str = "_") -> dict:
```
Comment (cbeddow) [inline mapillary/utils/format.py:384]: This is a good idea, nice work flattening it

File: mapillary/utils/format.py [hunk 2]
```diff
@@ -376,3 +379,205 @@ def geojson_to_polgyon(geojson: dict) -> dict:
...
+        {"coordinates": normalize_list(coordinate_list, width=width, height=height)}
+        if normalized
+        # Else return un-normalized coordinates
+        else {"coordinates": coordinate_list}
+    )
+
+
+def decode_pixel_geometry_in_geojson(
+    geojson: typing.Union[dict, GeoJSON],
+    normalized: bool = False,
```
Comment (cbeddow) [inline mapillary/utils/format.py:542]: Does this mean it will not normalize by default?

**Review by cbeddow — APPROVED** [review]

File: mapillary/utils/format.py (hunk 2 above)
Comment (Rubix982) [reply mapillary/utils/format.py:542]: No, by default it does not normalize. Should I change this to 'True' instead?

**Review by OmarMuhammedAli — APPROVED** [review]
Great Approach! LGTM

**Comment by Rubix982** [comment]
Thanks, @OmarMuhammedAli. :smile:

**Pull request merged**